plugins/modules/foreman_template_sync.py

```diff
@@ -0,0 +1,220 @@
+  force:
+    description: Update templates that are locked.
+    required: false
+    type: bool
+  lock:
+    description: Lock imported templates.
+    required: false
+    type: bool
+  branch:
+    description: Branch in Git repo.
+    required: false
+    type: str
+  repo:
+    description: Filesystem path or repo (with protocol), for example /tmp/dir or git://example.com/repo.git or https://example.com/repo.git.
+    required: false
+    type: str
+  filter:
+    description:
+      - On I(direction=import) import only templates with name matching this regular expression, after $prefix was applied.
+      - On I(direction=export) export templates with names matching this regex.
+      - Case-insensitive, snippets are not filtered.
+    required: false
+    type: str
+  negate:
+    description: Negate the prefix (for purging).
+    required: false
+    type: bool
+  dirname:
+    description: The directory within Git repo containing the templates.
+    required: false
+    type: str
+  locations:
+    description: REPLACE locations with given list.
+    required: false
+    type: list
+    elements: str
+  organizations:
+    description: REPLACE organizations with given list.
+    required: false
+    type: list
+    elements: str
+extends_documentation_fragment: foreman
+'''
+
+EXAMPLES = '''
+- name: Sync templates from git repo
+  foreman_template_sync:
+    repo: https://github.com/theforeman/community-templates.git
+    branch: 1.24-stable
+    associate: new
+    server_url: "https://foreman.example.com"
+    username: "admin"
+    password: "changeme"
+'''
+
+RETURN = ''' # '''
+
+from ansible.module_utils.foreman_helper import ForemanEntityAnsibleModule
+
+
+def main():
+    module = ForemanEntityAnsibleModule(
+        entity_spec=dict(
+            direction=dict(choices=['import', 'export'], default='import'),
+            location=dict(type='entity', flat_name='location_id'),
+            organization=dict(type='entity', flat_name='organization_id'),
+            associate=dict(choices=['always', 'new', 'never']),
+            metadata_export_mode=dict(choices=['refresh', 'remove', 'keep']),
+            prefix=dict(),
+            branch=dict(),
+            repo=dict(),
+            filter=dict(),
+            dirname=dict(),
+            verbose=dict(type='bool'),
+            force=dict(type='bool'),
+            lock=dict(type='bool'),
+            negate=dict(type='bool'),
+            locations=dict(type='entity_list', flat_name='location_ids'),
+            organizations=dict(type='entity_list', flat_name='organization_ids'),
+        ),
+        supports_check_mode=False,
+    )
+
+    entity_dict = module.clean_params()
+    module.connect()
+
+    if 'template' not in module.foremanapi.resources:
+        raise Exception('The server does not seem to have the foreman_templates plugin installed.')
+
+    all_templates = []
+
+    template_types = ['provisioning_templates', 'report_templates', 'ptables']
```

Comment:
This gets me thinking, whether we want to add a parameter to only affect a subset of available template types.

Reply:
This might be a good idea, but I have no control over it. The plugin will import whatever it finds. I need this only to compile a list of what's already present.
User might choose to incorporate the types into naming convention and then use filter. Or we fine a feature request to the plugin. I feel like I'll be doing this anyway e.g. for purging

Reply:
It was just a thought. But a feature request to the plugin sounds good anyway!